### PR TITLE
Move account/authorization connection step around

### PR DESCRIPTION
**Pull request by kdambekalns**
…

### CODE REVIEW

File: Classes/AssetSource/CantoAssetSource.php [hunk 1]
```diff
@@ -166,7 +166,7 @@ public function getLabel(): string
     }
 
     /**
-     * @return AssetProxyRepositoryInterface
+     * @return CantoAssetProxyRepository
```
Comment (kdambekalns) [inline Classes/AssetSource/CantoAssetSource.php:169]: Cleanup, unrelated to the actual change, but…

File: Classes/Controller/AuthorizationController.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    protected $accountAuthorizationRepository;
+
+    /**
+     * Finish OAuth2 authorization
+     *
+     * @throws OAuthClientException
+     * @throws StopActionException
+     * @throws UnsupportedRequestTypeException
+     */
+    public function finishAction(string $state, string $code, string $scope = ''): void
```
Comment (kdambekalns) [inline Classes/Controller/AuthorizationController.php:59]: A custom "finish authorization" action so we can redirect via HTML and not a `Location` header…

File: Classes/Controller/AuthorizationController.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Connect OAuth authorization to Neos account
+     *
+     * @throws StopActionException
+     * @throws UnsupportedRequestTypeException
+     * @throws IllegalObjectTypeException
+     * @throws SecurityException
+     */
+    public function connectAction(string $authorizationId, string $returnUri): void
```
Comment (kdambekalns) [inline Classes/Controller/AuthorizationController.php:77]: This connects the token to the Neos account

**Comment by kdambekalns** [comment]
I'll push some cleanup in a minute…

**Comment by kdambekalns** [comment]
Now this requires the (yet-to-be-released) new versions of https://github.com/flownative/flow-oauth2-client with [support for Authorization metadata](https://github.com/flownative/flow-oauth2-client/pull/29) but has less code and is more secure.

**Review by robertlemke — APPROVED** [review]
LGTM

**Pull request merged**